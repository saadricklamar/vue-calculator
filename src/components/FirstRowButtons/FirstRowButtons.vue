<template>
  <section class="first-row-buttons">
    <button class="ac">AC</button>
    <button class="plus-minus">+/-</button>
    <button class="percent">%</button>
    <button class="divide">/</button>
  </section>
</template>

<script>
export default {
  name: "FirstRowButtons"
};
</script>

<style>
button {
  height: 75px;
  width: 75px;
  margin: 2%;
  font-size: 1.5rem;
  border-radius: 5px;
  cursor: pointer;
  background-color: rgb(243, 241, 213);
}

button:hover {
 background-color: rgb(79,142,191);
 color: white;
}

.divide {
  background-color: #42b983;
  color: white;
}

.first-row-buttons {
  margin-top: 3%;
}
</style>