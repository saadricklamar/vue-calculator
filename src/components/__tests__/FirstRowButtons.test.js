import { mount } from "@vue/test-utils";
import FirstRowButtons from "../FirstRowButtons/FirstRowButtons";

describe("FirstRowButtons", () => {
  const wrapper = mount(FirstRowButtons, {
    propsData: {
      clearDisplay: jest.fn()
    }
  });
  it("should match the snapshot", () => {
    expect(wrapper.isVueInstance()).toMatchSnapshot();
  });
  it("should have the props of clearDisplay", () => {
    expect(wrapper.props('clearDisplay')).toBeTruthy();
  })
  it("should call clearDisplay when AC is clicked", () => {
    wrapper.vm.clearDisplay  = jest.fn();
    wrapper.find("button.ac").trigger("click");
    expect(wrapper.vm.clearDisplay).toHaveBeenCalled();
  });
});
