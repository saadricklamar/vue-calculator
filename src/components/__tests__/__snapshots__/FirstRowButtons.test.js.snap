// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstRowButtons should match the snapshot 1`] = `true`;
